rename `<async>` to `<system>` (#4421)
dfinity/motoko: src/lang_utils/error_codes/M0195.md

# M0195

> This warning means that you called a function that demands elevated (`system`) privileges,
> without manifestly passing the privilege.

Reviewer: ```suggestion
This warning means that you called a function that demands elevated (`system`) capabilities,
without manifestly passing the capability.
```

Dubious code example:

```motoko
Cycles.add(2_000_000_000_000);
```

If you encounter this warning, you should be giving some thought if the call can cause adverse behaviour,
and if you agree to perform the call you can add a manifest instantiation `<system>` (possibly followed
by further type parameters) to suppress the warning:

```motoko
Cycles.add<system>(2_000_000_000_000);
```

## Background

> Some builtin functions are defined to demand elevated privileges, as they perform system activities related

Reviewer: ```suggestion
Some built-in functions require additional system capabilities, as they perform privileged actions related
```

to resource management, involving values or privilege elevation. E.g. the builtin function

```motoko
setTimer : <system>(Nat64, Bool, () -> async ()) -> Nat
```

installs a callback which eventually gets invoked in an `async` context and as such can send messages and

> cycles. In order to avoid calls to `setTimer` from unprivileged synchronous code, its signature declares a demand

Reviewer: ```suggestion
cycles. In order to avoid calls to `setTimer` from unprivileged synchronous code, its signature requires system capability, represented by the pseudo type parameter 'system';
```

> `<system>`, which is normally not available in regular library code (unless resulting in `async` futures).

Reviewer: ```suggestion
`<system>`, which is normally not available in regular library code (unless in the body of a function that itself requiring `system` capability).
```

> The purpose of the annotation is to ensure an uninterruped, traceable chain of privilege passing to _dangerous_

Reviewer: ```suggestion
The purpose of the annotation is to provided an auditable trail of privilege passing to any potentially _dangerous_
```
